### PR TITLE
chore(goss): move windows sanity checks from provider script to goss

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3763
first step move windows sanity checks to goss

### CODE REVIEW

File: goss/goss-windows.yaml [hunk 1]
```diff
@@ -1,14 +1,107 @@
...
+    exit-status: 0
+    stdout:
+      - 1.4.0
+  chocolatey-awscli:
+    exec: C:\Program Files\Amazon\AWSCLIV2\aws.exe --version
+    exit-status: 0
+    stdout:
+      - 2.13.38
+  chocolatey-bundle:
+    exec: C:\tools\ruby26\bin\bundle.exe -v
```
Comment (smerle33) [inline goss/goss-windows.yaml:21]: need to test if I can remove all the full path and rely on the %PATH%

File: provisioning/windows-provision.ps1 [hunk 2]
```diff
@@ -417,8 +363,6 @@ Get-HotFix | Format-Table -Property HotFixID, Description, InstalledOn
 Write-Host "== Sanity Check of installed tools"
 Write-Host "- Path environment"
 Write-Host (Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name PATH).path
-Write-Host '- Sanity check for docker'
-& docker -v ## Client only
 foreach($k in $downloads.Keys) {
```
Comment (smerle33) [inline provisioning/windows-provision.ps1:366]: I think that this part could be removed

File: goss/goss-windows.yaml [hunk 3]
```diff
@@ -1,14 +1,107 @@
...
+    exec: C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\az.cmd version
+    exit-status: 0
+    stdout:
+      - 2.54.0
+  chocolatey:
+    exec: C:\ProgramData\Chocolatey\bin\choco.exe
+    exit-status: 0
+    stdout:
+      - 1.4.0
+  chocolatey-awscli:
```
Comment (dduportal) [inline goss/goss-windows.yaml:15]: Is there a reason to prefix with `chocolatey`?

Comment (smerle33) [reply goss/goss-windows.yaml:15]: except to point out the fact that it's installed by chocolatey none ... and will be best without planning the factorization ...

File: build-jenkins-agent-windows.pkr.hcl [hunk 4]
```diff
@@ -62,27 +62,42 @@ build {
     script            = "./provisioning/windows-provision.ps1"
   }
 
+  provisioner "powershell" {
```
Comment (smerle33) [inline build-jenkins-agent-windows.pkr.hcl:65]:
```suggestion
```
to remove

File: build-jenkins-agent-windows.pkr.hcl [hunk 5]
```diff
@@ -62,27 +62,42 @@ build {
...
     destination  = "C:/goss-windows.yaml"
   }
+
   provisioner "breakpoint" {
     note    = "Enable this breakpoint to pause before trying to run goss tests"
     disable = true
   }
+
   provisioner "powershell" {
+    # elevated_user     = local.windows_winrm_user[var.image_type]
```
Comment (smerle33) [inline build-jenkins-agent-windows.pkr.hcl:89]: remove comments

File: goss/goss-windows.yaml [hunk 6]
```diff
@@ -1,31 +1,163 @@
 command:
+  Powershell:
+    exec: powershell (Get-Host).Version
+    exit-status: 0
+  awscli:
+    exec: aws --version
+    exit-status: 0
+    stdout:
+      - 2.15.0
+  az:
```
Comment (smerle33) [inline goss/goss-windows.yaml:10]:
```suggestion
  azurecli:
```

File: goss/goss-windows.yaml [hunk 7]
```diff
@@ -1,31 +1,163 @@
...
+      - 2.55.0
+  bundle:
+    exec: bundle -v
+    exit-status: 0
+  chocolatey:
+    exec: choco
+    exit-status: 1
+    stdout:
+      - 1.4.0
+  cst:
```
Comment (smerle33) [inline goss/goss-windows.yaml:23]:
```suggestion
  container-structure-test:
```

File: goss/goss-windows.yaml [hunk 8]
```diff
@@ -1,31 +1,163 @@
...
+    stdout:
+      - 1.16.0
   default_java:
     exec: java --version
     exit-status: 0
     stdout:
       - 11.0.21+9
-  jdk8:
-    exec: C:\tools\jdk-8\bin\java.exe -version
+  docker:
```
Comment (smerle33) [inline goss/goss-windows.yaml:33]:
```suggestion
  docker-ce:
```

File: goss/goss-windows.yaml [hunk 9]
```diff
@@ -1,31 +1,163 @@
...
+    exec: docker -v
     exit-status: 0
-    stderr:
-      - 1.8.0_392
+  dockercompose:
```
Comment (smerle33) [inline goss/goss-windows.yaml:36]:
```suggestion
  docker_compose:
```

File: goss/goss-windows.yaml [hunk 10]
```diff
@@ -1,31 +1,163 @@
...
+    exec: docker-compose -v
+    exit-status: 0
+    stdout:
+      - 2.23.3
+  gh_cli:
+    exec: gh version
+    exit-status: 0
+    stdout:
+      - 2.40.0
+  gitlfs:
```
Comment (smerle33) [inline goss/goss-windows.yaml:46]:
```suggestion
  git_lfs:
```

File: goss/goss-windows.yaml [hunk 11]
```diff
@@ -1,31 +1,163 @@
...
+    stdout:
+      - 18.18.2
+  packer:
+    exec: packer --version
+    exit-status: 0
+    stdout:
+      - 1.9.4
+  pwsh:
+    exec: pwsh -command "(Get-Host).Version"
+    exit-status: 0
```
Comment (smerle33) [inline goss/goss-windows.yaml:119]:
```suggestion
    exit-status: 0
    stdout:
      - 7.4.0
```

**Review by smerle33 — COMMENTED** [review]
chore naming to match linux version

File: goss/goss-windows.yaml [hunk 12]
```diff
@@ -1,31 +1,163 @@
 command:
+  Powershell:
+    exec: powershell (Get-Host).Version
+    exit-status: 0
```
Comment (dduportal) [inline goss/goss-windows.yaml:4]: Not blocking: this is the default expected behavior in POSIX (and in `goss` of course). 
Can be removed to simplify (not in this PR though)

File: provisioning/windows-provision.ps1 (hunk 2 above)
Comment (dduportal) [reply provisioning/windows-provision.ps1:366]: Absolutely. As it should be done on a subsequent PR, let's do it in the updatecli PR ?

**Review by dduportal — APPROVED** [review]

**Comment by dduportal** [comment]
Build #22 is all green despite the GH check marked "in progress":

Capture d’écran 2023-12-14 à 15 16 38


As such, let's roll!

**Pull request merged**